### PR TITLE
Fix array handling

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Fixed
- How arrays are converted to string
- nested arrays support
- how null arrays are handled
- added struct type as string

### CODE REVIEW

File: src/metabase/driver/firebolt.clj [hunk 1]
```diff
@@ -101,23 +101,35 @@
...
    (database-type->base-type database-type))
 
-; Concatenate the elements of an array based on array elemets type (coverting array data type to string type to apply filter on array data)
-(defn is-string-array? [os]
-  (if (= (type (first (vec os))) java.lang.String) (str "['" (clojure.string/join "','" os) "']") (str "[" (clojure.string/join "," os) "]")))
-
-; Handle array data type
+(defn array-to-string [os]
+  (let [convert-element (fn [element]
+  (do (log/info "Class: " (.getName (.getClass element)) "is array" (.isArray (class element)))
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj]: let's remove this log

File: src/metabase/driver/firebolt.clj [hunk 2]
```diff
@@ -101,23 +101,35 @@
...
+                          (cond
+                            (.isArray (class element))
+                            (array-to-string (into [] element)) ; Handle nested arrays
+                            (instance? java.sql.Timestamp element)
+                            (.format (java.text.SimpleDateFormat. "yyyy-MM-dd HH:mm:ss") element)
+                            (instance? java.lang.String element)
+                            (str "'" element "'")
```
Comment (ptiurin) [inline src/metabase/driver/firebolt.clj:121]: do we support other types that require processing, like bytea? Also, how are other date types handled? Do we need another format here for `date` and `timestamptz`?

File: test/metabase/driver/firebolt_test.clj
Comment (ptiurin) [inline test/metabase/driver/firebolt_test.clj:1]: Let's also test null arrays and arrays containing nulls.

File: test/metabase/driver/firebolt_test.clj [hunk 3]
```diff
@@ -242,3 +243,56 @@
                                       (is (= [{:name "example_view"}]
                                              (filter (has-value :name "example_view") (map (partial into {})
                                                                                            (t2/select [Table :name] :db_id (u/the-id database))))))))))))
+
+; test that this method works
```
Comment (ptiurin) [inline test/metabase/driver/firebolt_test.clj]: let's remove this block before merging

**Comment by stepansergeevitch** [comment]
Знімок екрана 2024-12-23 о 5 36 04 пп

**Review by ptiurin — APPROVED** [review]
